File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void swap(int &r1,int &r2)
    {
        int t;
        t=r1;
        r1=r2;
        r2=t;
    }
    void moveZeroes(vector<int>& arr) {
        int n = arr.size();
        int zi=0;
        for(int i=0;i<n;i++)
        {
            if(arr[i] !=0)
            swap(arr[i],arr[zi++]);
        }
    }
};

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] arr) {
        int left = 0;
        int right = 0;
        for(int i=0;i<arr.length;i++)
        {
            if(arr[right]!=0)
            {
                int temp = arr[left];
                arr[left] = arr[right];
                arr[right] = temp;
                right++;
                left++;
            }
            else
            {
                right++;
            }
        }
    }
}